pkg/ingester/ingester_v2.go

```diff
@@ -204,13 +208,17 @@ func (i *Ingester) v2Query(ctx old_ctx.Context, req *client.QueryRequest) (*clie
 
 		it := series.Iterator()
 		for it.Next() {
+			numSamples++
```

Comment:
It would be more efficient to do `numSamples += len(ts.Samples)` after the loop.

Reply:
Right. Fixed both places.